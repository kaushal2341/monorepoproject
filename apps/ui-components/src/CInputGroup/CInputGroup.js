import React, {memo} from 'react';
import {InputGroup} from 'react-bootstrap';
import CInputGroupAppend from './CInputGroupAppend';
import CInputGroupPrepend from './CInputGroupPrepend';

const CInputGroup = (props) => {
    const {classes, id, as, bsPrefix, size, append, prepend, appendClassName, prependClassName} = props;
    return (
        <>
            <InputGroup
                className={classes}
                id={id}
                as={as}
                bsPrefix={bsPrefix}
                size={size}>
                {prepend ?
                    <CInputGroupPrepend
                        id={"prepend" + props.id}
                        className={appendClassName}
                        prepend={prepend}/>
                    : null}
                {append ?
                    <CInputGroupAppend
                        id={"append" + props.id}
                        className={prependClassName}
                        append={append}/>
                    : null}
            </InputGroup>
        </>
    )
};
export default memo(CInputGroup);