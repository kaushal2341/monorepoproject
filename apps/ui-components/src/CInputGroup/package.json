{
    "name": "@cogent/c-inputgroup",
    "version": "0.0.0",
    "main": "CInputGroup.js",
    "license": "MIT",
    "private": true
}