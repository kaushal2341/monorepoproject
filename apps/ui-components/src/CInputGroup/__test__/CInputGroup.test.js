import CInputGroup from '../CInputGroup';
import {InputGroup, Button} from 'react-bootstrap';

expect.addSnapshotSerializer(enzymeSerializer);
describe('InputGroup Test', () => {
    let wrapper, wrapper1;
    beforeEach(() => {
        wrapper = mount(<CInputGroup id="inputGroup"/>);
        wrapper1 = shallow(<CInputGroup id="inputGroup"/>);

    });
    afterEach(() => {
        wrapper.unmount();
    })
    it('should render successfully', () => {
        expect(wrapper1).toBeDefined();
    });
    it('should have only one input group component', () => {
        expect(wrapper1.find('#inputGroup')).toHaveLength(1);
    });
    it('should have given props available', () => {
        const propsAvailable = [
            'id',
            'className',
            'as',
            'size',
            'bsPrefix',
            'children'
        ];

        const propsOfInputGroup = Object.keys(wrapper.children().props());
        propsAvailable.map((propsA, index) => {
            expect(propsAvailable).toContain(propsOfInputGroup[index]);
        });
    });

    it('should have only one prepend input group element if provided', () => {
        const p = [
            <InputGroup.Text key="addOn-1" id="basic-addOn3">
                https://example.com/users/
            </InputGroup.Text>,
            <Button key="button2" id="button2" variant="outline-secondary">
                Button
            </Button>
        ];
        wrapper1.setProps({prepend: p});
        expect(wrapper1.find('#prependinputGroup')).toHaveLength(1)
    });

    it('should have only one append input group element if provided', () => {
        const a = [
            <InputGroup.Text key="addOn-1" id="basic-addOn3">
                https://example.com/users/
            </InputGroup.Text>,
            <Button key="button2" id="button2" variant="outline-secondary">
                Button
            </Button>
        ]
        wrapper1.setProps({append: a});
        expect(wrapper1.find('#appendinputGroup')).toHaveLength(1)
    });
    describe('Snapshot testing', () => {
        it('should match the previous snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })

    });
})