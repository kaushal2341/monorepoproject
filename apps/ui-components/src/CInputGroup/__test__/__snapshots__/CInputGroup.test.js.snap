// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Test Snapshot testing should match the previous snapshot 1`] = `
<Memo(CInputGroup)
  id="inputGroup"
>
  <Bootstrap(InputGroup)
    id="inputGroup"
  >
    <InputGroup
      bsPrefix="input-group"
      id="inputGroup"
    >
      <div
        className="input-group"
        id="inputGroup"
      />
    </InputGroup>
  </Bootstrap(InputGroup)>
</Memo(CInputGroup)>
`;
